### ISSUE TITLE
Automatically handling conflict between booking & timeoff event via API

### ISSUE THREAD

**Issue by Moristotle** [resourceguru/api-docs#68]
Hi, 

When creating and adding a timeoff event that conflicts with a booking (via API) the timeoff event is automatically added to a waiting list as an unconfirmed event that must be resolved manually from the dashboard.  

Is it possible to override the default behavior that conflicting timeoff events are automatically added to waiting list, and instead specify what will happen automatically happen with the timeoff when/if it encounters a conflict (when creating a timeoff event using the API endpoint)?  

The API docs does not mention anything about it and the documentation on Resource Guru focuses on how to resolve conflicts using the dashboard. 
https://help.resourceguruapp.com/en/articles/2942080-booking-clashes-the-waiting-list-overtime-and-capacity-planning

(Example of use case: automatically resolving conflicts by allowing the timeoff event to overwrite and/or delete the previous booking). - This way, it wouldn't be necessary to go to the platform and manually resolve every conflict.

**Comment by HendrikLouw** [comment]
Hi Moristotle,

You can use the following parameters to change the behaviour of the Downtimes POST endpoint:
* Setting `block_by_clashes: true` will return with status code `400` and return `clashes: [...]` with all the bookings that are clashing with the new downtime
* If you want to add clashing bookings to the waiting list automatically set `block_by_clashes: false, allow_waiting: true`
* If you want to delete clashing bookings automatically set `block_by_clashes: false, delete_invalid_bookings: true`

**Comment by jesperstarkar** [comment]
@HendrikLouw  if this in master? I can't find the `block_by_clashes` nor `delete_invalid_bookings` when searching for it.

**Comment by jesperstarkar** [comment]
And `delete_invalid_bookings` will that remove the new booking you try to make, or existing booking to make room for the new one? If existing, will it delete just a portion of the existing or the entire?

**Comment by Moristotle** [comment]
@HendrikLouw  Thank you very much, I got the intended result by passing block_by_clashes: false, and delete_invalid_bookings: true as part of the URL. 

For everyone that is following the discussion; this is how I did it: 
Made a post call to the downtimes endpoint with the following url: 
 <basepath>/v1/<account_id>/downtimes?block_by_clashes=false&delete_invalid_bookings=true. 

This lead to the timeoff event being added and the allocated time for the existing booking being removed/deleted for that specific time slot (where there was a clash). 
End result looks like this: 

![image](https://user-images.githubusercontent.com/43928778/78613930-65a06400-786d-11ea-9b0e-b6017765fb6e.png)
